### DISCUSSION TITLE
How to quickly go from insert to normal mode

### DISCUSSION THREAD

**Discussion by eminence** [helix-editor/helix#8345] [Q&A] [answered]
Hi all,

In vim, I use either <kbd>esc</kbd> or <kbd>ctrl+c</kbd> to instantly leave insert mode and go into normal mode.

In helix, I've found that pressing <kbd>esc</kbd> doesn't instantly bring be back to normal mode.  Instead I have to wait a few hundred milliseconds.  I think this is because the terminal is waiting to see if there's another keypress, because pressing <kbd>esc</kbd> and then another key is the same as using the <kbd>alt</kbd> modifier.  

My bash terminal has this same timeout behavior, but somehow vim disables this.  Does anyone have any hints about how to get helix to do the same thing?

**Comment by the-mikedavis** [comment, marked as answer]
Are you using tmux or screen? See https://github.com/helix-editor/helix/wiki/Troubleshooting#when-using-tmux-or-screen-there-is-a-delay-after-hitting-escape-before-its-registered

**Reply by eminence** [reply]
I am, yes!  I'll try that setting, thanks.  Sorry for the noise, I thought I searched the FAQ, I can't explain why I didn't find that entry
